network-mux/src/Network/Mux/Bearer/Socket.hs

```diff
@@ -61,8 +62,8 @@ socketAsMuxBearer tracer sd = do
           --say "read"
           --hexDump hbuf ""
           case Mx.decodeMuxSDU hbuf of
-              Left  e      -> throwM e
-              Right header -> do
+              Left  e       -> throwM e
+              Right !header -> do
```

Comment:
Wouldn't it be better to ensure that `Mx.decodeMuxSDU` returns a result that is in normal form?